integration/integration_test.go

```diff
@@ -552,13 +552,21 @@ func TestBuildFromDockerfile(t *testing.T) {
 			"Dockerfile": "FROM " + testImageAlpine,
 		},
 	})
-	ctr, err := runEnvbuilder(t, runOpts{env: []string{
-		envbuilderEnv("GIT_URL", srv.URL),
-		envbuilderEnv("DOCKERFILE_PATH", "Dockerfile"),
-		envbuilderEnv("DOCKER_CONFIG_BASE64", base64.StdEncoding.EncodeToString([]byte(`{"experimental": "enabled"}`))),
-	}})
+	logbuf := new(bytes.Buffer)
+	ctr, err := runEnvbuilder(t, runOpts{
+		env: []string{
+			envbuilderEnv("GIT_URL", srv.URL),
+			envbuilderEnv("DOCKERFILE_PATH", "Dockerfile"),
+			envbuilderEnv("DOCKER_CONFIG_BASE64", base64.StdEncoding.EncodeToString([]byte(`{"experimental": "enabled"}`))),
+			"DOCKER_CONFIG=/config", // Ignored, because we're setting DOCKER_CONFIG_BASE64.
+		},
+		logbuf: logbuf,
+	})
 	require.NoError(t, err)
 
+	require.Contains(t, logbuf.String(), "Set DOCKER_CONFIG to /.envbuilder")
+	require.NotContains(t, logbuf.String(), "Using DOCKER_CONFIG at ")
```

Comment:
Do we generally trust log output for test assertions?

If this log line ever drifts, this test will become a silent false positive. Is there a practical way to assert against the actual behaviour of the system? Could we copy the written docker config secret file so that it makes its way into the container and then assert against its contents?

Would that approach be worth it?

Reply:
We could modify `ENVBUILDER_INIT_SCRIPT` to write the content of `$DOCKER_CONFIG`

Reply:
@SasSwart good points, I agree, in general it’s best to test behavior and not output. @johnstcn I wanted to do that here but we can’t do it in the init script since it runs after cleanup.

Not sure if it’ll work but we could try to copy it as part of Dockerfile build instructions, will look into that.

Reply:
Went back to the drawing board with the implementation, changed the behavior a bit for user-provided config files and converted the integration tests to primarily testing behavior.